staging/src/k8s.io/apiextensions-apiserver/test/integration/helpers.go

```diff
@@ -92,3 +93,81 @@ func updateCustomResourceDefinitionWithRetry(client clientset.Interface, name st
 	}
 	return nil, fmt.Errorf("too many retries after conflicts updating CustomResourceDefinition %q", name)
 }
+
+// getCRDSchemaForVersionOrDie returns the validation schema for given version in given CRD.
+func getCRDSchemaForVersionOrDie(crd *apiextensionsv1beta1.CustomResourceDefinition, version string) *apiextensionsv1beta1.CustomResourceValidation {
```

Comment:
prefer propagating errors to taking down the server

Reply:
The error should never happened here. Earlier in the serveHTTP function we make sure the requestVersion is valid

Reply:
Also this will not bring down the server. I am sure somewhere in http handler we catch panics and fail the request.

Reply:
My question is why it is preferable when the error is impossible to happened. There is no scenareo this error can be happened. If it happened, these are two options:
- Return a soft error which fail the request with something user won't understand (logically this cannot happened, so there is no good explanation for the end user for this)
- Return internal error with a panic logged into the api server logs which get attention from admins and ultimately us.

The chance of a panic getting our attention is more and the end result for the user is the same so I would prefer panic here. Also If we don't prefer panic here, I would say no other DoXOrDie make sense.